hw/ip/otbn/rtl/otbn_model.cc

```diff
@@ -0,0 +1,141 @@
+#include "svdpi.h"
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <fcntl.h>
+#include <sstream>
+#include <iostream>
+#include <string>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+// Candidate for hw/dv/verilator/cpp
+/**
+ * Guard class for SV Scope
+ *
+ * This guard ensures that all functions in the context where it is instantiated
+ * are executed in an SVScope. It will restore the previous scope at destruction
+ * and thereby make sure that the previous scope will be restored for all paths
+ * that exit the scope.
+ */
+class SVScoped {
+  svScope m_prevScope = 0;
+  SVScoped() {}
```

Comment:
Our style guide requires explicit `public:` and `private:` in classes. Also, public members come before private members.